### PR TITLE
test(CB2-6125): improve VTM test coverage

### PR DESCRIPTION
**Pull request by tahayazici**
## Improve VTM Test Coverage

Attain 100% test coverage (statements, branches, functions & lines) for the following files:

1. app.component.ts
2. breadcrumbs.components.ts
3. global-error.service.ts
4. app-role-required.directive.ts
5. search.component.ts
6. tech-record.component.ts
7. edit-tech-record-button.component.ts
8. tech-record-summary.component.ts
9. vehicle-technical-record.component.ts

[CB2-6125](https://dvsa.atlassian.net/browse/CB2-6125)

### CODE REVIEW

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts [hunk 1]
```diff
@@ -50,37 +56,50 @@ export class VehicleTechnicalRecordComponent implements OnInit, AfterViewInit {
...
-    forms.forEach(form => {
-      DynamicFormService.updateValidity(form, errors);
-    });
 
-    errors.length > 0 ? this.errorService.setErrors(errors) : this.errorService.clearErrors();
+    forms.forEach(form => DynamicFormService.updateValidity(form, errors));
+
+    errors.length > 0
+      ? this.errorService.setErrors(errors)
+      : this.errorService.clearErrors();
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:78]: Minor detail, but you can just do this
```suggestion
    errors.length
      ? this.errorService.setErrors(errors)
      : this.errorService.clearErrors();
```

**Review by gjulien-bjss — APPROVED** [review]

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts (hunk 1 above)
Comment (tahayazici) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts]: Done

**Review by gjulien-bjss — APPROVED** [review]

File: src/app/features/search/search.component.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export class SearchComponent {
     return errors.find(error => error.anchorLink === name);
   }
 
-  public get Roles() {
+  public get roles() {
```
Comment (naathanbrown) [inline src/app/features/search/search.component.ts:40]: I think we need to standardise this across the repo, I recently swapped one of these the other way, an annoying job but if you have the time would be great, if not ignore this comment!

File: src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts [hunk 3]
```diff
@@ -66,23 +32,31 @@ export class EditTechRecordButtonComponent implements OnInit {
...
       this.toggleEditMode();
-    });
+    }
+
+    this.errorService.clearErrors();
+    this.store.dispatch(updateEditingTechRecordCancel());
+  }
+
+  submitTechRecord() {
+    this.submitCheckFormValidity.emit();
```
Comment (naathanbrown) [inline src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts:58]: How come we are removing the logic here? I thought this was needed?

File: src/app/features/search/search.component.ts (hunk 2 above)
Comment (naathanbrown) [reply src/app/features/search/search.component.ts:40]: You already did this lower down lol

File: src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts (hunk 3 above)
Comment (naathanbrown) [reply src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts:58]: Oh this logic has been moved to the parent?

**Review by Twebb97 — APPROVED** [review]

File: src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts (hunk 3 above)
Comment (tahayazici) [reply src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts:58]: Yep

**Review by naathanbrown — DISMISSED** [review]

**Review by gjulien-bjss — COMMENTED** [review]
Tests failing and loads of _can't bind to x_ errors in console

**Review by naathanbrown — APPROVED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Pull request merged**